### PR TITLE
Added Tegile Storage OS detection, storage pool, and project storage.

### PR DESCRIPTION
**Pull request by theherodied**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #7921

### CODE REVIEW

**Comment by theherodied** [comment]
Don't approve this yet. It's a work in progress. I just want testers to try it.

**Comment by theherodied** [comment]
This is ready to merge. Tegile OS detection and pool usage has been added. 
More will be added in a future PR.

**Comment by laf** [comment]
Thanks for this @theherodied.

Is tegile the correct os name? It should be based on the firmware / OS it's actually running so should this be IntelliFlash instead?

**Comment by theherodied** [comment]
@laf, I went back and forth on that from what I saw in the disco and poller. I can update it but how would you want it named?

They call it Tegile IntelliFlash in the snmp output. 

Just intelliflash all lowercase?

**Comment by theherodied** [comment]
@laf, let me know how this is now.
I didn't change the icon name as it's really the Tegile Logo I believe. I also left the mib directory as tegile/TEGILE-MIB. I left the mib named as they have it in the actual mib file itself.

**Comment by theherodied** [comment]
Changed array to intelliflash1. Intelliflash2 will be for project usage as it has to be calculated differently.

File: includes/definitions/intelliflash.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+text: 'Tegile IntelliFlash'
+type: storage
+icon: tegile
+over:
+    - { graph: device_storage, text: 'Storage Usage' }
+mib_dir:
+    - tegile
+discovery:
+    - sysDescr:
+      - Tegile IntelliFlash
```
Comment (laf) [inline includes/definitions/intelliflash.yaml]: Looks like you can use `sysObjectID` instead for detection. From our stats `.1.3.6.1.4.1.43906.1` is unique to this device. sysObjectID is more stable for detection.

File: includes/discovery/storage/intelliflash.inc.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.storage snmpgnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2018 Ryan Finney
+ * @author     https://github.com/theherodied/
+ */
+use LibreNMS\Config;
+if($device['os'] = 'intelliflash') {
```
Comment (laf) [inline includes/discovery/storage/intelliflash.inc.php]: `=` will always match here :) Should be `==`

**Comment by laf** [comment]
Test data updated, hopefully that fixes travis.

**Comment by murrant** [comment]
Storage units is not set (you need to set it to 1) and the SQL is failing, that is why tests fail.

**Comment by theherodied** [comment]
Fixes: #7921

**Pull request merged**